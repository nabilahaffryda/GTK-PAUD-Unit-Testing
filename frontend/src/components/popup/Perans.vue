<template>
  <v-dialog v-model="dialog" scrollable max-width="400px" persistent>
    <v-card>
      <v-card-title class="headline">Ubah Peran Anda</v-card-title>
      <v-divider />
      <v-card-text>
        <v-list flat dense>
          <template v-for="(item, i) in roles">
            <v-list-item class="grey--text" :key="i" @click="onSwitch(item)">
              <v-list-item-icon><v-icon>mdi-swap-horizontal</v-icon></v-list-item-icon>
              <v-list-item-title class="grey--text text--darken-2">
                {{ item && item.label }}
              </v-list-item-title>
            </v-list-item>
          </template>
        </v-list>
      </v-card-text>
      <v-divider />
      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn color="blue darken-1" text @click.native="dialog = false">
          Batal
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>
<script>
import { mapActions } from 'vuex';
export default {
  data() {
    return {
      dialog: false,
      roles: [
        { key: 'instansi', label: 'Admin Instansi' },
        { key: 'gtk', label: 'GTK' },
      ],
    };
  },
  methods: {
    ...mapActions('preferensi', ['getPreferensi']),

    open() {
      this.dialog = true;
    },

    onSwitch(role) {
      this.dialog = false;
      this.$store.commit('auth/SET_ROLE', role.key);
      window.location.href = '/';
    },
  },
};
</script>
