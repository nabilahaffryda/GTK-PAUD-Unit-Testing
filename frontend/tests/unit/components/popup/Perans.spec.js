import { mount, createLocalVue } from "@vue/test-utils";
import Perans from "@/components/popup/Perans.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('Perans.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })

    test('render popup peran', () => {
        const wrapper = mount(Perans, {
            localVue,
            vuetify,
            data() {
                return {
                    roles: [
                        { key: 'instansi', label: 'Admin Instansi' },
                        { key: 'gtk', label: 'GTK' },
                    ],
                };
            },
        });
        expect(wrapper).toMatchSnapshot();
    });
})
