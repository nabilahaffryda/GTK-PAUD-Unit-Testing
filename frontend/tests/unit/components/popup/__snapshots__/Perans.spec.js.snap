// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Perans.vue render popup peran 1`] = `
<div role="dialog" class="v-dialog__container">
  <!---->
</div>
`;
